### PR TITLE
[RFC] Add support for DirectCounter extern to PSA/eBPF backend

### PR DESCRIPTION
**Pull request by tatry**
This PR adds the support for DirectCounter extern to PSA/eBPF backend.

Co-authored-by: Tomasz Osiński <tomasz@opennetworking.org>
Co-authored-by: Mateusz Kossakowski <mateusz.kossakowski@orange.com>

### CODE REVIEW

File: backends/ebpf/tests/p4testdata/counters.p4 [hunk 1]
```diff
@@ -56,11 +56,16 @@ control ingress(inout headers hdr,
...
+        action_cnt.count((bit<32>)hdr.ethernet.etherType);
+        send_to_port(ostd, egress_port);
+    }
+
     table tbl_fwd {
         key = {
             istd.ingress_port : exact;
         }
-        actions = { do_forward; NoAction; }
+        actions = { do_forward; do_forward_2; NoAction; }
```
Comment (osinstom) [inline backends/ebpf/tests/p4testdata/counters.p4:68]: why do we need the second `do_forward` ?

File: backends/ebpf/ebpfTable.h [hunk 2]
```diff
@@ -104,6 +104,7 @@ class EBPFTable : public EBPFTableBase {
     virtual void emitValueType(CodeBuilder* builder);
     virtual void emitValueActionIDNames(CodeBuilder* builder);
     virtual void emitValueStructStructure(CodeBuilder* builder);
+    virtual void emitDirectTypes(CodeBuilder* builder) { (void) builder; }
```
Comment (osinstom) [inline backends/ebpf/ebpfTable.h]: Why do we put `emitDirectTypes` in a classic EBPF table? I'm not sure if there is a plan for `ebpf_model` to implement DirectCounters..

File: backends/ebpf/tests/p4testdata/counters.p4 (hunk 1 above)
Comment (tatry) [reply backends/ebpf/tests/p4testdata/counters.p4:68]: Just another way to call a Counter method in an action - with action argument and with header field. I used this to check if this also compiles correctly.

File: backends/ebpf/ebpfTable.h (hunk 2 above)
Comment (tatry) [reply backends/ebpf/ebpfTable.h]: It is called inside `emitValueType`, so moving to PSA table it would require rewrite also that function.

**Review by osinstom — APPROVED** [review]

**Pull request closed without merging**